matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerImporter.java

```diff
@@ -0,0 +1,469 @@
+                .setX(mBranch.getX() * zb)
+                .setG(0)
+                .setB(mBranch.getB() / zb)
+            .add();
+    }
+
+    private static void createBranches(MatpowerModel model, ContainersMapping containerMapping, Network network, PerUnitContext perUnitContext) {
+        for (MBranch mBranch : model.getBranches()) {
+            if (mBranch.getRatio() == 0) {
+                createLine(mBranch, containerMapping, network, perUnitContext);
+            } else {
+                createTransformer(mBranch, containerMapping, network, perUnitContext);
+            }
+        }
+    }
+
+    @Override
+    public String getFormat() {
+        return FORMAT;
+    }
+
+    @Override
+    public List<Parameter> getParameters() {
+        return Collections.singletonList(IGNORE_BASE_VOLTAGE_PARAMETER);
+    }
+
+    @Override
+    public String getComment() {
+        return "MATPOWER Format to IIDM converter";
+    }
+
+    private String findExtension(ReadOnlyDataSource dataSource, boolean throwException) throws IOException {
+        for (String ext : EXTENSIONS) {
+            if (dataSource.exists(null, ext)) {
+                return ext;
+            }
+        }
+        if (throwException) {
+            throw new MatpowerException("File " + dataSource.getBaseName()
+                    + "." + String.join("|", EXTENSIONS) + " not found");
+        }
+        return null;
+    }
+
+    @Override
+    public boolean exists(ReadOnlyDataSource dataSource) {
+        try {
+            String ext = findExtension(dataSource, false);
+            return dataSource.exists(null, ext);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public void copy(ReadOnlyDataSource fromDataSource, DataSource toDataSource) {
+        Objects.requireNonNull(fromDataSource);
+        Objects.requireNonNull(toDataSource);
+        try {
+            String extFrom = findExtension(fromDataSource, false);
+            String extTo = findExtension(toDataSource, false);
+            try (InputStream is = fromDataSource.newInputStream(null, extFrom);
+                 OutputStream os = toDataSource.newOutputStream(null, extTo, false)) {
+                ByteStreams.copy(is, os);
+            }
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    @Override
+    public Network importData(ReadOnlyDataSource dataSource, NetworkFactory networkFactory, Properties parameters) {
+        Objects.requireNonNull(dataSource);
+        Objects.requireNonNull(networkFactory);
+        Network network = networkFactory.createNetwork(dataSource.getBaseName(), FORMAT);
+
+        // no info abount time & date from the matpower file, set a  default
+        ZonedDateTime caseDateTime = DEFAULTDATE.atStartOfDay(ZoneOffset.UTC.normalized());
+        network.setCaseDate(new DateTime(caseDateTime.toInstant().toEpochMilli(), DateTimeZone.UTC));
```

Comment:
See above, the default date is now. For round trip, you have to set-up a post-processor maybe to fix the date. I'm pretty sure this is the reason why you fix the date!?

Reply:
done: I've fixed the date in the test, before the check takes place.